### DISCUSSION TITLE
Use Laravel Sanctum for vuejs

### DISCUSSION THREAD

**Discussion by smzapp** [Laravel-Backpack/community-forum#1240] [Q&A (Help)]
I have been using laravel backpack for my project using default bootstrap jquery. And then, it uses web session-based authentication. Using laravel backpack, how could I use Laravel Sanctum for the authentication in order to protect my VUEJS app.

https://stackoverflow.com/questions/79214747/use-sanctum-in-laravel-backpack-authentication

[a comment by an automated account is not shown]

**Comment by karandatwani92** [comment]
Hey @smzapp 

Backpack's authentication uses a completely separate authentication driver, provider, guard and password broker. They're all named `backpack`.

If you need a separate login/auth for the front user, go ahead and set up Sanctum regularly.

I also want to highlight that [Laravel Sanctum](https://laravel.com/docs/11.x/sanctum) supports both
 - Token-based auth
 - and "stateful" authentication using Laravel session cookies.
 
 I'm using it on my project(Laravel+Sanctum+Backpack+lighthouse-php(graphQL API)). But yes, the sanctum has a learning curve. 
 I choose to keep it stateful, so I don't need to bother about key storing and protecting on the client side. AFAIR, The following two `.env` attributes helped to make it stateful
 ```
 SESSION_DOMAIN=.get-set-sold.test
SANCTUM_STATEFUL_DOMAINS=.get-set-sold.test
```
